dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/RuleActionAssignValueImplementer.java

```diff
@@ -44,12 +44,14 @@
 import org.springframework.stereotype.Component;
 
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.transaction.annotation.Transactional;
 
 /**
  * @Author Zubair Asghar.
  */
 @Slf4j
 @Component( "org.hisp.dhis.programrule.engine.RuleActionAssignValueImplementer" )
+@Transactional
```

Comment:
We standardized the usage of `@Trasactional`: it should be used on methods rather that at class level. I have the impression that this method does not really need this annotation, since all the methods are 

1. read-only 
2. calling other methods on different classes that are already annotated. 

Please review the remaining methods and move the @Transactional annotation only to methods where it is needed.

Reply:
fixed